Sources/SourceKitLSPAPI/BuildDescription.swift

```diff
@@ -160,14 +169,52 @@ public struct BuildDescription {
 
     /// The inputs of the build plan so we don't need to re-compute them  on every call to
     /// `fileAffectsSwiftOrClangBuildSettings`.
-    private let inputs: [BuildPlan.Input]
+    private let inputs: [Build.BuildPlan.Input]
 
-    // FIXME: should not use `BuildPlan` in the public interface
+    /// Wrap an already constructed build plan.
     public init(buildPlan: Build.BuildPlan) {
         self.buildPlan = buildPlan
         self.inputs = buildPlan.inputs
     }
 
+    /// Construct a build description, compiling build tool plugins and generating their output when necessary.
+    public static func load(
+        destinationBuildParameters: BuildParameters,
+        toolsBuildParameters: BuildParameters,
+        packageGraph: ModulesGraph,
+        pluginConfiguration: PluginConfiguration,
+        traitConfiguration: TraitConfiguration,
+        disableSandbox: Bool,
+        scratchDirectory: URL,
+        fileSystem: any FileSystem,
+        observabilityScope: ObservabilityScope
+    ) async throws -> (description: BuildDescription, errors: String) {
+        let bufferedOutput = BufferedOutputByteStream()
+        let threadSafeOutput = ThreadSafeOutputByteStream(bufferedOutput)
+
+        // This is quite an abuse of `BuildOperation`, building plugins should really be refactored out of it. Though
+        // even better would be to have a BSP server that handles both preparing and getting settings.
+        // https://github.com/swiftlang/swift-package-manager/issues/8287
+        let operation = BuildOperation(
+            productsBuildParameters: destinationBuildParameters,
+            toolsBuildParameters: toolsBuildParameters,
+            cacheBuildManifest: true,
+            packageGraphLoader: { packageGraph },
+            pluginConfiguration: pluginConfiguration,
+            scratchDirectory: try Basics.AbsolutePath(validating: scratchDirectory.path),
+            traitConfiguration: traitConfiguration,
+            additionalFileRules: FileRuleDescription.swiftpmFileTypes,
+            pkgConfigDirectories: [],
+            outputStream: threadSafeOutput,
+            logLevel: .error,
+            fileSystem: fileSystem,
+            observabilityScope: observabilityScope
+        )
+
+        let plan = try await operation.generatePlan()
+        return (BuildDescription(buildPlan: plan), bufferedOutput.bytes.description)
```

Comment:
question: The single test added validate a single `load` call, more specifically, the `description`, which is of type `BuildDescription`. 

-  Do other tests exist which validate `BuildOperation.generatePlan()` in isolation?
- Do other tests exists that validate `bufferedOutput.bytes.description` is populated as expected based on the inputs to `BuildOperation` and the action `generatePlan()`?

The test is only validation a single case for the `BuildDescription`.  Can we add a test that validate the `error: String` , or do other tests validate the

Reply:
`generatePlan` only has package visibility and is an internal implementation detail, I don't see a point adding a test for it.

`outputStream` is necessarily tested basically everywhere since it's how we output anything. Adding a test for it would just be testing BufferedOutputByteStream/ThreadSafeOutputByteStream, which seems pointless.